src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/IModelBindingContext.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding
+{
+    /// <summary>
+    /// A context that contains operating information for model binding and validation.
+    /// </summary>
+    public interface IModelBindingContext
+    {
+        /// <summary>
+        /// Gets or sets a model name which is explicitly set using an <see cref="IModelNameProvider"/>.
+        /// <see cref="Model"/>.
+        /// </summary>
+        string BinderModelName { get; set; }
+
+        /// <summary>
+        /// Gets the <see cref="Type"/> of an <see cref="IModelBinder"/> associated with the
+        /// <see cref="Model"/>.
+        /// </summary>
+        Type BinderType { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value which represents the <see cref="BindingSource"/> associated with the
```

Comment:
`ModelBinding.BindingSource` since properties bind more tightly than types. (The `ModelBindingContext` class happens to contain a lot of incorrect comment links.)